sdks/python/apache_beam/ml/inference/pytorch_inference.py

```diff
@@ -40,11 +41,32 @@
 def _load_model(
     model_class: torch.nn.Module, state_dict_path, device, **model_params):
   model = model_class(**model_params)
-  model.to(device)
+
+  if device == torch.device('cuda') and not torch.cuda.is_available():
+    logging.warning(
+        "Model handler specified a 'GPU' device, but GPUs are not available. " \
+        "Switching to CPU.")
+    device = torch.device('cpu')
+
   file = FileSystems.open(state_dict_path, 'rb')
-  model.load_state_dict(torch.load(file))
+  try:
+    logging.info(
+        "Loading state_dict_path %s onto a %s device", state_dict_path, device)
+    state_dict = torch.load(file, map_location=device)
+  except RuntimeError as e:
+    message = "Loading the model onto a GPU device failed due to an " \
```

Comment:
Are we certain that if there is a runtime error this is the reason?  Could this message ever be a red herring?

Reply:
Well, I can't say that the error message will always be: `Attempting to deserialize object on a CUDA device but torch.cuda.is_available() is False`. Be it will definitely stem from the `torch.load(file, map_location=device)` call. So that should provide enough context as to where the exception has occured.